hibernate-core/src/main/java/org/hibernate/type/CharacterArrayNClobType.java

```diff
@@ -29,4 +38,23 @@ public String getName() {
 		return null;
 	}
 
+	@Override
+	public <X> BasicType<X> resolveIndicatedType(
+			JdbcTypeDescriptorIndicators indicators,
+			JavaTypeDescriptor<X> domainJtd) {
+		if ( domainJtd != null && domainJtd.getJavaTypeClass() == char[].class ) {
+			// domainJtd is a `char[]` instead of a `Character[]`....
+			final TypeConfiguration typeConfiguration = indicators.getTypeConfiguration();
+			final JdbcTypeDescriptorRegistry jdbcTypeRegistry = typeConfiguration.getJdbcTypeDescriptorRegistry();
+			final JdbcTypeDescriptor jdbcType = jdbcTypeRegistry.getDescriptor( Types.NCLOB );
+
+			return typeConfiguration.getBasicTypeRegistry().resolve(
+					typeConfiguration.getJavaTypeDescriptorRegistry().getDescriptor( char[].class ),
```

Comment:
Shouldn't this use `Character[]`?

```suggestion
					typeConfiguration.getJavaTypeDescriptorRegistry().getDescriptor( Character[].class ),
```

Reply:
Or maybe just pass the `domainJtd`?